src/main.sh

```diff
@@ -161,7 +161,7 @@ nix_eval_args=()
 nix_eval() {
   local output_type=$1; shift
   local nix=$1; shift
-  nix eval "$output_type" "(
+  nix eval "$output_type" --impure --expr "(
```

Comment:
This would make execution fail with the non-flake version. However if `nix-prefetch` is wrapped to have the stable nix in its `$PATH` than https://github.com/msteen/nix-prefetch/pull/8
is sufficient even if the nix-daemon is the flake one.

Reply:
Ahh, that was evident to me right away. It seems the expression in this repo and in nixpkgs has stable nix in it's path. 
I actually PR'd https://github.com/msteen/nix-prefetch/pull/11/commits/4b5373e0a96ada9318759d215ab6f4336ceb0dd3 so that the stable nix wouldn't be preferred over the version of nix used in the system.

Reply:
I might be missing something, but these flags will always break that call to `nix` for all non-flake versions of nix, regardless of where on PATH it is, would it not?

Reply:
For example:
```
> nix --version
nix (Nix) 2.3.2
> nix eval --expr '(1+1)'
error: unrecognised flag '--expr'
```

Reply:
Yes, you need to wait until nix 2.4 or newer become the new nix.